### ISSUE TITLE
Add Docker instructions to Readme

### ISSUE THREAD

**Issue by cdevroe** [cdevroe/unmark#267]
The appropriate docker-compose files have been part of the repository and release build for months. After having moved between several systems and several platforms; Mac and Windows mostly, I've worked out most of the kinks I've personally come across.

So I'll be adding the instructions to the readme and will consider Docker fully supported as a way to run Unmark locally. In fact, it is the preferred way!

**Comment by runlevel-six** [comment]
This is likely because of something on my end, but with a fresh clone of `unmark`, using either the `trunk` or `proposals/v2020.3` branch, with no existing docker volume, when I run `docker-compose up -d` and then navigate to http://localhost/setup (or just http://localhost) I get the following errors:

```
Warning: mysqli::real_connect(): (HY000/2002): No such file or directory in /var/www/html/system/database/drivers/mysqli/mysqli_driver.php on line 203

Warning: Cannot modify header information - headers already sent by (output started at /var/www/html/system/database/drivers/mysqli/mysqli_driver.php:203) in /var/www/html/system/core/Common.php on line 570

A Database Error Occurred
Unable to connect to your database server using the provided settings.

Filename: controllers/Install.php

Line Number: 8
```

That hasn't been an error I've seen when not running it containerized - figured I'd drop this here in case it was something outside of my experience whose fix is just an additional step in the instructions.

**Comment by cdevroe** [comment]
@runlevel-six Thanks for reporting the issue. I _just_ added Unmark to a new computer and I followed the Readme instructions and didn't get this issue. I wonder if you may have an old config (which isn't replaced by downloading or upgrading Unmark)?

**Comment by runlevel-six** [comment]
@cdevroe thanks for the response. 

It was a fresh install on a new Ubuntu 20.04 server (I am planning to migrate from an existing install on an older version of Ubuntu). I just cloned the repository, ran the docker-compose up command, and then got this error. I then stopped the stack, removed the containers and volume, checked out the `proposals/v2020.3` branch, and tried it again.

Honestly, it sounds like I did the same thing as you. I'll dig a little deeper. It does sound like in general the instructions are fine as is from your testing, so I don't want to hijack the thread - I can create a new issue if I don't find a fix. If I do find a fix I'll update this just to tie up the loose end (especially if it does turn out relevant to the process).

**Comment by cdevroe** [comment]
@runlevel-six I can also quickly destroy my local set up and try again as well. I'll see if I can do this very soon. Sorry you're having issues. The Docker container should be the _easiest_ way to use Unmark locally.

**Comment by cdevroe** [comment]
@runlevel-six I've found the issue, I believe. The database-sample.php file does not have the proper credentials for the included database using Docker Compose. So the error you're receiving is very misleading. I'm unsure how to correct that error - but I have been able to fix the issue which will be included in 2020.3 which will be released publicly today.

**Comment by runlevel-six** [comment]
@cdevroe that did the trick. Everything worked fine from a fresh install following the instructions. Thanks!
